src/main/java/org/sagebionetworks/migration/utils/ToolMigrationUtils.java

```diff
@@ -42,18 +42,17 @@ public static List<TypeToMigrateMetadata> buildTypeToMigrateMetadata(boolean isS
 		return l;
 	}
 
-	
-	public static List<MigrationTypeCountDiff> getMigrationTypeCountDiffs(List<MigrationTypeCount> srcCounts, List<MigrationTypeCount> destCounts) {
-		List<MigrationTypeCountDiff> result = new LinkedList<MigrationTypeCountDiff>();
-		Map<MigrationType, Long> mapSrcCounts = new HashMap<MigrationType, Long>();
-		for (MigrationTypeCount sMtc: srcCounts) {
-			mapSrcCounts.put(sMtc.getType(), sMtc.getCount());
-		}
+	public static List<MigrationTypeMetaDiff> getMigrationTypeMetaDiffs(List<MigrationTypeCount> srcCounts, List<MigrationTypeCount> destCounts) {
+		List<MigrationTypeMetaDiff> result = new LinkedList<MigrationTypeMetaDiff>();
 		// All migration types of source should be at destination
 		// Note: unused src migration types are covered, they're not in destination results
-		for (MigrationTypeCount mtc: destCounts) {
-			MigrationTypeCountDiff outcome = 	new MigrationTypeCountDiff(mtc.getType(), (mapSrcCounts.containsKey(mtc.getType()) ? mapSrcCounts.get(mtc.getType()) : null), mtc.getCount());
-			result.add(outcome);
+		for (MigrationTypeCount destMtc: destCounts) {
+			MigrationTypeMetaDiff metaDiff = null;
+			Optional<MigrationTypeCount> found = srcCounts.stream().filter(mtc->mtc.getType().equals(destMtc.getType())).findFirst();
```

Comment:
You can first build a map from the source migration type to the MigrationTypeCount:

`Map<MigrationType, MigrationTypeCount> sourceTypeMap = source.stream().collect(Collectors.toMap(MigrationTypeCount::getType, Function.identity()));
		
		dest.forEach( destCount -> {
			MigrationTypeCount sourceCount = sourceTypeMap.get(destCount.getType());
			
			if (sourceCount != null) {
				result.add(new MigrationTypeMetaDiff(destCount.getType(), new MetaReport(sourceCount.getMinid(), destCount.getMinid()), new MetaReport(sourceCount.getMaxid(), destCount.getMaxid())));					
			}
			
			
		});`